specifyweb/frontend/js_src/lib/components/WbPlanView/mappingHelpers.ts

```diff
@@ -75,14 +94,44 @@ export const formattedEntry = `${schema.fieldPartSeparator}formatted`;
  */
 export const emptyMapping = '0';
 
+/**
+ * Returns a complete tree definition name from a tree definition name
+ * (e.x Taxononomy => %Taxonomy)
+ * Opposite of getNameFromTreeDefinitionName
+ */
+export const formatTreeDefinition = (definitionName: string): string =>
+  `${schema.treeDefinitionSymbol}${definitionName}`;
+
 /**
  * Returns a complete tree rank name from a tree rank name
  * (e.x Kingdom => $Kingdom)
  * Opposite of getNameFromTreeRankName
  *
  */
 export const formatTreeRank = (rankName: string): string =>
-  `${schema.treeSymbol}${rankName}`;
+  `${schema.treeRankSymbol}${rankName}`;
+
+// Delimiter used for rank name keys i.e: <rankname>~><treeId>
+const RANK_KEY_DELIMITER = '~>';
+
+/**
+ * Returns a formatted tree rank name along with its tree id: (e.x $Kingdom => $Kingdom~>1)
+ * Used for generating unique key names when there are multiple trees with the same rank name.
+ * Opposite of getRankNameWithoutTreeId
+ * See: https://github.com/specify/specify7/pull/5091#issuecomment-2328037741
+ */
+export const formatTreeRankWithTreeId = (
+  rankName: string,
+  treeId: number
+): string => `${rankName}${RANK_KEY_DELIMITER}${treeId}`;
```

Comment:
it might be more intuitive to have tree id before the rank name
after all, rank name is inside a tree, not the other way around

is the a reason to do it otherwise?

Reply:
Would something like the following be a little more intuitive and require the same amount of work as the current implementation to parse correctly? 

```json
"Taxonomy~>Subspecies": { 
       "treeNodeCols": { 
         "name": "Subspecies", 
         "author": "Subspecies Author" 
       }, 
       "treeId": 1 
     }, 
"1~>Subspecies": { # or alternatively
       "treeNodeCols": { 
         "name": "Subspecies", 
         "author": "Subspecies Author" 
       }, 
     }, 
```

Reply:
I like the look of `<treename>~><rankname>`. Should be a simple enough change on the frontend
```JSON
"Taxonomy~>Subspecies": { 
       "treeNodeCols": { 
         "name": "Subspecies", 
         "author": "Subspecies Author" 
       }, 
       "treeId": 1 
     }, 
```
Would it break already existing upload plans on the backend? I think maybe a reason we went with rankname~>treeId was to be as backwards compatible as possible?

Reply:
>  I think maybe a reason we went with rankname~>treeId was to be as backwards compatible as possible?

Agree with Jason, we shouldn't take a crappy design just because it "looks" similar to previous one